lib/spack/spack/hash_types.py

```diff
@@ -51,12 +50,19 @@ def attr(self):
     name='process_hash'
 )
 
-#: Full hash used in build pipelines to determine when to rebuild packages.
-full_hash = SpecHashDescriptor(
-    deptype=('build', 'link', 'run'), package_hash=True, name='full_hash')
 
-
-#: Package hash used as part of full hash
+#: Package hash used as part of dag hash
 package_hash = SpecHashDescriptor(
     deptype=(), package_hash=True, name='package_hash',
     override=lambda s: s.package.content_hash())
+
+
+# Deprecated hash types, no longer used, but needed to understand old serialized
+# spec formats
```

Comment:
Do we already tell users when we'll deprecate being able to read old formats in? That would be good for us to know when we can remove the remaining references to `build_hash` and `full_hash`

Reply:
At the moment, I don't know and I'm inclined not to deal with this, as `Spec` is very backward compatible.  I do think we should do this at some point.